### PR TITLE
Update CMake with recent changes, remove more borland checks

### PR DESCRIPTION
**Pull request by MaartenBent**

### CODE REVIEW

File: samples/nativdlg/dialog1.rc [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 #include "resource.h"
 
-#if !defined(__GNUWIN32__) && !defined(__BORLANDC__)
+#if !defined(__GNUWIN32__)
```
Comment (vadz) [inline samples/nativdlg/dialog1.rc:24]: I have no idea why are we testing for this one here...

**Review by vadz — APPROVED** [review]
Thanks!

I'll merge this once the CI builds finish, just to be sure.

File: samples/nativdlg/dialog1.rc (hunk 1 above)
Comment (MaartenBent) [reply samples/nativdlg/dialog1.rc:24]: All code in the sample looks 20 years old. I don't plan to rewrite it but I can change this one to `#include <windows.h>`, like other .rc files.
I also noticed the label is not shown in the button. I'll fix that as well.
nativdlg

Comment (MaartenBent) [reply samples/nativdlg/dialog1.rc]: This is how it looks like after 1a2d813
nativdlg-fixed

**Pull request merged**